### ISSUE TITLE
Pass variables in lists

### ISSUE THREAD

**Issue by turnerrainer** [buerokratt/Ruuter#160]
AS A Service Manager
I WANT TO pass variables in lists the same way it's done in all other cases
SO THAT I could make such REST requests where necessary

### Acceptance Criteria

- [ ] The following example DSL works

> Currently using `${currentYear.response.body.year}` as a variable doesn't work

#### Example DSL

```yml
currentYear:
  call: http.get
  args:
    url: https://www.timeapi.io/api/Time/current/zone?timeZone=Europe/Tallinn
  result: currentYear

correctYear:
  call: http.post
  args:
    url: https://andmed.stat.ee/api/v1/et/stat/RV021
    body:
      query:
        - code: Sugu
          selection:
            filter: item
            values:
              - 1
        - code: Aasta
          selection:
            filter: item
            values:
              - ${currentYear.response.body.year}
        - code: Vanuserühm
          selection:
            filter: item
            values:
              - 1
      response:
        format: json-stat2
  result: SAanswer
```

**Comment by RayDNoper** [comment]
Currently this DSL assignes variable correctly, the GET returns:
`
{"year":2023,"month":6,"day":6,"hour":13,"minute":56,"seconds":9,"milliSeconds":428,"dateTime":"2023-06-06T13:56:09.4288008","date":"06/06/2023","time":"13:56","timeZone":"Europe/Tallinn","dayOfWeek":"Tuesday","dstActive":true}
`

and the generated POST body is:
`
{"query":"[{code=Sugu, selection={filter=item, values=[1]}}, {code=Aasta, selection={filter=item, values=[2023]}}, {code=Vanuserühm, selection={filter=item, values=[1]}}]","response":{"format":"json-stat2"}}
`

Apparently the stat.ee site responds to this query with "Bad Request".

**Comment by turnerrainer** [comment]
Turned out to be a bug described in https://github.com/buerokratt/Ruuter/discussions/166 and fixed in #167
